Ginger/Ginger/Drivers/DriversWindows/MobileDriverWindow.xaml.cs

```diff
@@ -190,6 +317,302 @@ private async void MobileDriverWindow_DriverMessageEvent(object sender, DriverMe
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("realDisplaySize", out value))
+                {
+                    if (string.IsNullOrEmpty((string)value))
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Screen resolution:", "N/A", DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Screen resolution:", (string)value, DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("displayDensity", out value))
+                {
+                    if (string.IsNullOrEmpty(value.ToString()))
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Display density:", "N/A", DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Display density:",value.ToString(), DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("carrierName", out value))
+                {
+                    if (string.IsNullOrEmpty((string)value))
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Carrier name:", "N/A", DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Carrier name:", (string)value, DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                }
+
+                if (mDeviceGeneralInfo.TryGetValue("bluetooth", out value))
+                {
+                    if (((Dictionary<string, object>)value).Count > 0)
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Bluetooth:", ((Dictionary<string, object>)value)["state"].ToString(), DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                    else
+                    {
+                        mDeviceDetails.Add(new DeviceInfo("Bluetooth:", "N/A", DeviceInfo.eDeviceInfoCategory.Detail));
+                    }
+                }
+
+                if (mDeviceBatteryInfo.TryGetValue("power", out battery))
+                {
+                    mDeviceDetails.Add(new DeviceInfo("Device Battery:", battery + "%", DeviceInfo.eDeviceInfoCategory.Detail));
+                }
+                else
+                {
+                    mDeviceDetails.Add(new DeviceInfo("Battery:", "N/A", DeviceInfo.eDeviceInfoCategory.Detail));
+                }
+
+            }
+            else
+            {
+                mDeviceDetails.Add(new DeviceInfo("Device ID:", "N/A", DeviceInfo.eDeviceInfoCategory.Detail));
```

Comment:
No need to maintain 2 "N/A" set, just add to the IF check both checks if field exist and not null